public/app/plugins/datasource/prometheus/language_provider.ts

```diff
@@ -81,22 +87,10 @@ export default class PromQlLanguageProvider extends LanguageProvider {
     return [];
   };
 
-  start = () => {
-    if (!this.startTask) {
-      this.startTask = this.fetchMetrics();
-    }
-    return this.startTask;
-  };
-
-  fetchMetrics = async () => {
-    this.metrics = await this.fetchMetricNames();
+  start = async (): Promise<any> => {
```

Comment:
I assume this works because of the `any` but the issue is that LanguageProvider specifies it as `abstract start: () => Promise<any[]>;`. So if somebody was using this as a `LanguageProvider` type they could assume something iterable, while getting just undefined here and the code would fail.